lib/commands/sdk.js

```diff
@@ -549,427 +573,524 @@ SdkSubcommands.install = {
-					return appc.version.gte(v, '3.0.0');
-				}).sort().reverse(),
-				newestInstalledSdk = config.get('sdk.selected', config.get('app.sdk', 'latest')) === 'latest' ? (vers.length ? vers[0] : null) : config.get('sdk.selected', config.get('app.sdk')),
-				analyticsPayload = {};
-
-			data.version && (analyticsPayload.version = data.version);
-			data.branch && (analyticsPayload.branch = data.branch);
-
-			function finish(version) {
-				analyticsPayload.version = version;
-				cli.addAnalyticsEvent('sdk.install', analyticsPayload);
-
-				if (version) {
-					var setDefault = (data.setDefault || cli.argv.default) && appc.version.gte(version, '3.0.0') && (cli.argv.default || newestInstalledSdk === null || appc.version.gt(version, newestInstalledSdk));
-					if (setDefault) {
-						logger.log(__('Setting Titanium SDK %s as the default.', version.cyan) + '\n');
-						config.load();
-						config.set('sdk.selected', version);
-						config.save();
-					}
 
-					logger.log(__('Titanium SDK %s successfully installed!', version.cyan) + '\n');
-				} else {
-					logger.log(__('Successfully installed!') + '\n');
+				if (bar) {
+					bar.tick(total);
+					this.logger.log('\n');
 				}
 
-				finished();
-			}
-
-			if (data.file) {
-				// Avoid extracting the whole SDK if we already have an SDK with the same version and timestamp installed
-				if (!cli.argv.force) {
-					// Pull out the version and os from the zipfile name!
-					var m = data.file.match(/^mobilesdk-(\d+\.\d+\.\d+(?:\.\w+)?)-(osx|win32|linux)\.zip$/);
-					if (m) {
-						try {
-							// if entry doesn't exist, we get back empty string, which throws error for JSON.parse
-							var manifest = new AdmZip(data.file).readAsText('mobilesdk/' + m[3] + '/' + m[1] + '/manifest.json'),
-								json = manifest ? JSON.parse(manifest) : null,
-								version = json ? (json.name || json.version) : null;
-							// if we got the manifest and the version matches and the timestamp matches, assume it's already installed and don't re-install
-							if (version && cli.env.sdks[version] && cli.env.sdks[version].manifest.timestamp === json.timestamp) {
-								logger.log(__('Titanium SDK %s is already installed!', version.cyan) + '\n');
-								process.exit(0);
-							}
-						} catch (e) {
-							logger.error(__('Failed to extract SDK version from zipfile %s', data.file.cyan) + '\n');
-							process.exit(1);
-						}
-					}
+				// get the actual version number
+				const manifestFile = path.join(installLocation, 'manifest.json');
+				if (fs.existsSync(manifestFile)) {
+					const manifest = fs.readJSONSync(manifestFile);
+					this.version = manifest.version || this.version;
 				}
-				extractSDK(data.file, true, null, finish);
-			} else if (data.url) {
-				downloadSDK(data.url, function (filename) {
-					extractSDK(filename, cli.argv['keep-files'], data.version, finish);
-				});
-			}
+
+				resolve();
+			});
 		});
 	}
-};
+
+	/**
+	 * @param {object} sdks
```

Comment:
<!--
  0 failure: 
  1 warning:  lib/commands/sdk....
  
  
  DangerID: danger-id-Danger;
  File: lib/commands/sdk.js;
  Line: 1069;
-->

- :warning: lib/commands/sdk.js line 1069 – Missing JSDoc parameter description for 'sdks'. (valid-jsdoc)